test cases/unit/35 dist script/subprojects/sub/meson.build

```diff
@@ -8,4 +9,7 @@ else
   # The dist script replace prog.c with something that actually build.
   meson.add_dist_script('dist-script.py', 'success')
   executable('prog', 'prog.c')
+
+  versiontest = find_program('version-test.py')
+  test('dist version replacement', versiontest, args : meson.project_version())
```

Comment:
Given that all tests on this PR pass without #12782 being merged, I guess I'm not testing correctly.